### PR TITLE
Calendar: Clicking on today should not change selection

### PR DESCRIPTION
**Pull request by ElianaLiberman**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Include a change request file using `$ npm run change`

#### Description of changes

When clicking on "Today", don't select the date and close the calendar. 
We still allow the option of having this behavior if desired but not by default any more.

#### Focus areas to test

(optional)

### CODE REVIEW

**Comment by cliffkoh** [comment]
Before you check in, could you please edit the title since the title becomes the commit message? :)

File: common/changes/office-ui-fabric-react/elliberm-gototoday_2018-07-11-19-51.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+  "changes": [
+    {
+      "packageName": "office-ui-fabric-react",
+      "comment": "don't select date by defaultick on Today",
```
Comment (cliffkoh) [inline common/changes/office-ui-fabric-react/elliberm-gototoday_2018-07-11-19-51.json]: Before you check in, could you please fix this to be a proper changelog entry? (This gets recorded/published into the release notes https://github.com/OfficeDev/office-ui-fabric-react/releases)

Something like: Calendar: [......]

File: packages/office-ui-fabric-react/src/components/Calendar/Calendar.tsx [hunk 2]
```diff
@@ -299,19 +299,26 @@ export class Calendar extends BaseComponent<ICalendarProps, ICalendarState> impl
...
 
-  private _onGotoToday = (): void => {
+  private _onGotoToday = (selectDateOnClick: boolean): void => {
     const { dateRangeType, firstDayOfWeek, today, workWeekDays } = this.props;
 
     const dates = getDateRangeArray(today!, dateRangeType!, firstDayOfWeek!, workWeekDays!);
 
-    this._onSelectDate(today!, dates);
+    if (selectDateOnClick) {
+      this._onSelectDate(today!, dates);
```
Comment (cliffkoh) [inline packages/office-ui-fabric-react/src/components/Calendar/Calendar.tsx]: I realise there is a lot of preexisting usages of "!" in line 305, but could we try to minimize it (such that we comment all usages of it providing a justification)?

Why are we certain that `today` will never be undefined, if it is presumably an optional prop?

**Comment by dzearing** [comment]
@lorejoh12 Can you please review this?

File: packages/office-ui-fabric-react/src/components/Calendar/Calendar.tsx [hunk 3]
```diff
@@ -299,19 +299,28 @@ export class Calendar extends BaseComponent<ICalendarProps, ICalendarState> impl
     }
   };
 
-  private _onGotoToday = (): void => {
+  private _onGotoToday = (selectDateOnClick: boolean): void => {
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/Calendar/Calendar.tsx]: Hmm, this is weird up to a click handler. The first parameter of a click handler is an event args. So... wouldn't selectDateOnClick always be truthy?

File: packages/office-ui-fabric-react/src/components/Calendar/Calendar.tsx [hunk 4]
```diff
@@ -299,19 +299,28 @@ export class Calendar extends BaseComponent<ICalendarProps, ICalendarState> impl
...
 
+  private _onGotoTodayClick = (ev: React.MouseEvent<HTMLElement>): void => {
+    this._onGotoToday(false /*selectDateOnClick*/);
+  };
+
   private _onGotoTodayKeyDown = (ev: React.KeyboardEvent<HTMLElement>): void => {
     if (ev.which === KeyCodes.enter) {
       ev.preventDefault();
-      this._onGotoToday();
+      this._onGotoToday(false /*selectDateOnClick*/);
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/Calendar/Calendar.tsx]: Is onGotoToday ever called with true? I was unclear on why the prop was needed.

There aren't unit tests validating this. Possible to add some?

**Comment by dzearing** [comment]
@lorejoh12 :)

**Pull request merged**